Example/Firebase.xcodeproj/project.pbxproj

```diff
@@ -5057,6 +5056,13 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
+				HEADER_SEARCH_PATHS = (
+					"$(SRC_ROOT)/../Firebase/**",
+					"$(SRC_ROOT)/../Firestore/**",
```

Comment:
I don't think this is valid. The Firebase project shouldn't be referring to any Firestore headers; certainly not any internal ones.

Reply:
It should be noted that this isn't the core Firebase SDK. These were added so that the autocomplete in Xcode would work as expected in the tests, but given that there are no Firestore or Functions tests in this project, I do think they should be removed.

This comment also made the realize that developers might copy this project, which could lead to inadvertent use of internal headers, so the other search paths should only be added to test targets rather than the project as a whole.

Thanks for pointing this out! I made [PR 2123](https://github.com/firebase/firebase-ios-sdk/pull/2123) to revert this change. I'll follow up with a more limited version of this that only affects test targets and only includes headers that are used in those tests.

Reply:
It should be noted that this isn't the core Firebase SDK. These were added so that the autocomplete in Xcode would work as expected in the tests, but given that there are no Firestore or Functions tests in this project, I do think they should be removed.

This comment also made the realize that developers might copy this project, which could lead to inadvertent use of internal headers, so the other search paths should only be added to test targets rather than the project as a whole.

Thanks for pointing this out! I made [PR 2123](https://github.com/firebase/firebase-ios-sdk/pull/2123) to revert this change. I'll follow up with a more limited version of this that only affects test targets and only includes headers that are used in those tests.

Reply:
+1 for using specific header paths and avoiding recursive search paths.